### ISSUE TITLE
Does not show variables in completion but functions in C file

### ISSUE THREAD

**Issue by Schievel1** [universal-ctags/citre#181]
## Describe the bug
When in a C file I get completion candidates for functions just fine, but I don't get suggestion for variables. 

Here is a small test example. I use the global backend and the `mytext` variable is not suggested from corfu:
![grafik](https://github.com/user-attachments/assets/7ccaa714-2be4-4117-a5e0-98b3bc108cb9)
on the other hand it completes the function `myInBufferFunction`:
![grafik](https://github.com/user-attachments/assets/d8518044-cb78-4ca9-b429-77433fdc1e14)
(don't mind the header file, I was just testing around if it has something to do with whether the function is defined in-buffer or in another file - it does not) 

Heres my config regarding citre:
```
(use-package! citre
  :demand t
  :init
  (require 'citre-config) ;; needed for lazy load
  :config
  (setq
   citre-peek-auto-restore-after-jump nil
   citre-global-completion-case-sensitive nil
   citre-project-root-function #'projectile-project-root
   citre-default-create-tags-file-location 'global-cache
   citre-use-project-root-when-creating-tags t)
  (evil-define-key 'normal citre-peek-keymap
    (kbd "]t") #'citre-peek-next-definition
    (kbd "[t") #'citre-peek-prev-definition
    (kbd "]p") #'citre-peek-through
    (kbd "[p") #'citre-peek-through-references
    (kbd "[j") #'citre-peek-jump
    (kbd "ESC") #'citre-peek-abort
    )
  (evil-make-overriding-map citre-peek-keymap 'motion)
  (evil-make-overriding-map citre-peek-keymap 'normal))
```

## To Reproduce

If you are **pretty sure** about the cause, or you are **pretty sure** it's a easy fix, or you are **pretty sure** it can be easily reproduced by developers, you can delete this section. Otherwise, write steps to reproduce, and it should start by `$ emacs -Q`.

Steps to reproduce:

1. `$ emacs -Q`
2.  I can not really reproduce this with emacs -Q. I need to load corfu and compat and citre and whatnot and it does not show me completions at all. 

## System information

- OS: Gentoo Linux in WSL2
- Emacs version: 29.4
- Are you using the latest version of Citre on master branch: I am using the latest form melpa.

## Additional context

I tried with ggtags-mode and it is showing the variables, so it must be a setting or something in citre.

**Comment by Schievel1** [comment]
fwiw the GNU Global project seems to know about this it says here: https://www.gnu.org/software/global/plans.html

> Treat variable definitions as a definition tag (GTAGS).

**Comment by Schievel1** [comment]
Yup, it works fine when using ctags to throught global when creating the database. I did this to let it use ctags:
`  (setenv "GTAGSLABEL" "ctags")`

I don't know if that is really the right way to go.

**Comment by AmaiKinono** [comment]
Hi. You could look into the gtags config file `gtags.conf`. Our [document](https://github.com/universal-ctags/citre/blob/master/docs/user-manual/use-global-backend.md#more-tips-on-configuring-gnu-global) also mentions that.

A gtags "label" is just a set of gtags settings in `gtags.conf`, including which parser to use for scanning definitions and references. `GTAGSLABEL` variable specifies which label to use. So yeah, if you like the behavior of bundled `ctags` label, setting the environment variable is the right thing to do. If you want more, you can customize your own label.

**Comment by Schievel1** [comment]
It is not just a set of settings that is named 'ctags' coincidentally. Global is _using_ the ctags binary with that settings when the label is set to ctags. 
I had to compile Global with `--with-universal-ctags=/usr/bin/exuberant-ctags` to tell it were the ctags executable is.

**Comment by AmaiKinono** [comment]
> It is not just a set of settings that is named 'ctags' coincidentally. 

It is if you look into it. The default config file on my system is`/usr/share/gtags/gtags.conf`, the `ctags` label is defined as:

```
ctags:\
	:tc=exuberant-ctags:tc=htags:
...
exuberant-ctags|plugin-example|setting to use Exuberant Ctags plug-in parser:\
	:tc=common:\
	:ctagscom=/usr/bin/etags:\
...
	:gtags_parser=Ant\:$ctagslib:\
... gtags parser of other languages are also set to ctagslib. That's why "Global is *using* the ctags binary"
```

See `man gtags.conf` to know the syntax. The location of ctags executable can be modified by the `ctagscom` setting.

> `➜ global --color=never --encode-path=' :' --result=grep --literal --reference --symbol -- mytext`

This is the command Citre use to find references, not definitions or completions. But from the output, a possible guess is the line `Warning: source file './test.c' is not available.` can not be parsed by Citre, so it runs into an error.

I'd suggest the following test to know exactly where does the problem happen.

1. Run
 
   ```
   $ global --completion --color=never --encode-path=' :' --result=grep --literal -- my
   ```

   to see if `global` could find the completions.

2. Eval `(citre-global--get-lines "my" 'completion)` to see if the global backend actually gets the output of `global`.
3. Eval `(citre-global-get-tags "my" 'completion (not citre-global-completion-case-sensitive) 'alpha)` to see if Citre can parse the output.
4. Eval `(citre-global-get-completions)` with the point after a typed "my" to see if the global completion backend works properly.
5. Eval `(citre-get-backend-and-completions)` to see if the `global` backend is actually used.

**Comment by Schievel1** [comment]
> global --completion --color=never --encode-path=' :' --result=grep --literal -- my

```
➜ global --completion --color=never --encode-path=' :' --result=grep --literal -- my
myInBufferFunction
myOtherFileFunction
```
so it does not get them. I continue with the other points regardless.

> Eval (citre-global--get-lines "my" 'completion) to see if the global backend actually gets the output of global.

```
("myInBufferFunction" "myOtherFileFunction")
```

> Eval (citre-global-get-tags "my" 'completion (not citre-global-completion-case-sensitive) 'alpha) to see if Citre can parse the output.

```
(#s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myInBufferFunction")) #s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myOtherFileFunction")))
```

> (citre-global-get-completions)

```
(208 210 (#s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myInBufferFunction")) #s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myOtherFileFunction"))))
```

> (citre-get-backend-and-completions)

```
(global 208 210 (#s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myInBufferFunction")) #s(hash-table size 16 test eq rehash-size 1.5 rehash-threshold 0.8125 data (name "myOtherFileFunction"))))
```

But with  `(setenv "GTAGSLABEL" "ctags")` I get

```
➜ global --completion --color=never --encode-path=' :' --result=grep --literal -- my
myInBufferFunction
myOtherFileFunction
myglobal
mytext
```

**Comment by AmaiKinono** [comment]
> Btw I found this [1] and isn't this the same problem we have here?

I think yes.

> Maybe run once with and once without `--symbols` then concatenate the results or something?

Yes. I've tested this and it's doable. But although this is a improvement for auto-completion, we still cannot find definitions of such symbols using global. So at the end, a global user may still want to use the ctags plugin parser.

**Comment by AmaiKinono** [comment]
I'd like to close this as it's a problem of global, and the suggested fix doesn't make a whole lot difference (finding definitions will still not work.
